cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2863,70 +2863,57 @@ private boolean isCustomCertUpToDate(StatefulSet sts, Pod pod, String annotation
-
-            if (log.isDebugEnabled()) {
-                List<String> reasons = new ArrayList<>();
-                for (Ca ca: cas) {
-                    if (ca.certRenewed()) {
-                        reasons.add(ca + " certificate renewal");
-                    }
-                    if (ca.certsRemoved()) {
-                        reasons.add(ca + " certificate removal");
-                    }
-                    if (!isPodCaCertUpToDate(pod, ca)) {
-                        reasons.add("Pod has old " + ca + " certificate generation");
-                    }
-                }
-                if (isAncillaryCmChange) {
-                    reasons.add("ancillary CM change");
+                if (ca.certRenewed()) {
+                    reasons.add(ca + " certificate renewal");
                 }
-                if (!isPodUpToDate) {
-                    reasons.add("Pod has old generation");
-                }
-                if (isFsResizeNeeded)   {
-                    reasons.add("file system needs to be resized");
-                }
-                if (!isCustomCertTlsListenerUpToDate)   {
-                    reasons.add("custom certificate on the TLS listener changes");
-                }
-                if (!isCustomCertExternalListenerUpToDate)   {
-                    reasons.add("custom certificate on the external listener changes");
+                if (ca.certsRemoved()) {
+                    reasons.add(ca + " certificate removal");
                 }
                 if (nodeCertsChange) {
-                    reasons.add("server certificates changed");
+                    reasons.add(ca + " certificate changed");
                 }
-                if (!reasons.isEmpty()) {
-                    if (isPodToRestart) {
-                        log.debug("{}: Rolling pod {} due to {}",
-                                reconciliation, pod.getMetadata().getName(), reasons);
-                    }
+                if (!isPodCaCertUpToDate(pod, ca)) {
+                    reasons.add("Pod has old " + ca + " certificate generation");
+                }
+            }
+            if (isAncillaryCmChange) {
+                reasons.add("ancillary CM change");
+            }
+            if (!isPodUpToDate(sts, pod)) {
+                reasons.add("Pod has old generation");
+            }
+            if (fsResizingRestartRequest.contains(pod.getMetadata().getName()))   {
+                reasons.add("file system needs to be resized");
+            }
+            if (!isCustomCertUpToDate(sts, pod, KafkaCluster.ANNO_STRIMZI_CUSTOM_CERT_THUMBPRINT_TLS_LISTENER))   {
+                reasons.add("custom certificate on the TLS listener changes");
+            }
+            if (!isCustomCertUpToDate(sts, pod, KafkaCluster.ANNO_STRIMZI_CUSTOM_CERT_THUMBPRINT_EXTERNAL_LISTENER))   {
+                reasons.add("custom certificate on the external listener changes");
+            }
```

Comment:
The old code didn't do this. It used the `isCustomCertTlsListenerUpToDate` field. Why change it?